code/modules/vending/_vending.dm

```diff
@@ -827,6 +834,10 @@ GLOBAL_LIST_EMPTY(vending_products)
 	if(!active)
 		return
 
+	if(COOLDOWN_FINISHED(src, product_ad_cooldown) && LAZYLEN(small_ads) > 0)
```

Comment:
I dont think you can have a negative list

Reply:
you can have a list of length 0

Reply:
> you can have a list of length 0

Which translates into something funny in byond logic (it turns into FALSE)